spec/integration/rom/factory_spec.rb

```diff
@@ -243,10 +243,114 @@
       expect(jane.email).to eql('jane@doe.org')
 
       expect(john.first_name).to eql('John')
+      expect(john.last_name).to eql('Doe')
       expect(john.email).to eql('john@doe.org')
     end
   end
 
+  context 'with traits' do
+    it 'allows to define traits' do
+      factories.define(:user) do |f|
+        f.timestamps
+
+        f.trait :jane do |t|
+          t.first_name 'Jane'
+          t.email 'jane@doe.org'
+        end
+
+        f.trait :doe do |t|
+          t.last_name 'Doe'
+        end
+      end
+
+      jane = factories.structs[:user, :jane]
+
+      expect(jane.first_name).to eql('Jane')
+      expect(jane.last_name).to eql nil
+      expect(jane.email).to eql('jane@doe.org')
+
+      jane_doe = factories.structs[:user, :jane, :doe]
+
+      expect(jane_doe.first_name).to eql('Jane')
+      expect(jane_doe.last_name).to eql('Doe')
+      expect(jane_doe.email).to eql('jane@doe.org')
+    end
+
+    it 'allows to define nested traits' do
+      factories.define(:user) do |f|
+        f.timestamps
+
+        f.trait :jane do |t|
+          t.first_name 'Jane'
+          t.email 'jane@doe.org'
+        end
+
+        f.trait :jane_doe, %i[jane] do |t|
+          t.last_name 'Doe'
+        end
+      end
+
+      jane = factories.structs[:user, :jane_doe]
+
+      expect(jane.first_name).to eql('Jane')
+      expect(jane.last_name).to eql('Doe')
+      expect(jane.email).to eql('jane@doe.org')
+    end
+
+    it 'allows to define traits for persisted' do
```

Comment:
what is this test for?

Reply:
[Persistable](https://github.com/rom-rb/rom-factory/pull/18/files#diff-8b6c33d5b26232d866a729facfb35a68) - is a different builder.

Reply:
Oh, I just didn't notice the difference between this test and the few above.
Now I see - there are `factories.struct[]` and `factories[]` calls